src/realm/object-store/impl/collection_notifier.hpp

```diff
@@ -209,9 +202,34 @@ class CollectionNotifier {
     std::function<bool(ObjectChangeSet::ObjectKeyType)> get_modification_checker(TransactionChangeInfo const&,
                                                                                  ConstTableRef);
 
+    // Returns a vector containing all `KeyPathArray`s from all `NotificationCallback`s attached to this notifier.
+    std::vector<KeyPathArray> get_key_path_arrays();
+    /**
+     * Collects all `ColKey`s for all `m_callback`s attached to this notifier. These keys can then be used to compare
+     * against the actual changes made to decide wheather or not we need to send notifications based on the key path
+     * filtered keys.
+     *
+     * @param only_for_root_table If set to true, the `ColKey`s will only be gather for the root table.
+     *
+     * @return A `std::vector` of all `ColKey`s in all `m_callback`s.
+     */
+    std::vector<ColKey> get_filtered_column_keys(bool only_for_root_table);
```

Comment:
This function doesn't seem to make any sense? The ColKeys from a bunch of different tables aren't something that you can do anything useful with, as columns in different tables can have the same ColKey.

Reply:
Thanks for pointing this out! I wrongly assumed the ColKey to be unique.
Fixed with 55e0a71.